File: Tema 4/AA 4.2 Reporte práctico/riegoIoT.pl
% Estado de sensores
humedad_suelo(baja).
temperatura(35).
hora(20).
pronostico_lluvia(false).

% Regla: ¿Es una hora adecuada para regar?
hora_adecuada :- hora(H), (H < 10 ; H > 18).

% Regla principal: ¿Cuándo se debe activar el sistema de riego?
activar_riego :- 
    humedad_suelo(baja), 
    pronostico_lluvia(false), 
    hora_adecuada.

% Diagnóstico del sistema
estado_riego('Activado') :- activar_riego.
estado_riego('No activado') :- \+ activar_riego.

% Regla para alerta de temperatura extrema
alerta_calor :- 
    temperatura(T), 
    T >= 32.

% Mensaje de alerta si se activa el riego con calor extremo
recomendacion :-
    activar_riego,
    alerta_calor, !,
    writeln('Alerta: Riego activado con temperatura alta. Considere riego nocturno o por goteo.').

recomendacion :-
    writeln('Sin recomendaciones especiales para el riego.').

File: Tema 4/AA 4.2 Reporte práctico/riegoIoT.py
# Estado de sensores
humedad_suelo = "baja"
temperatura = 35
hora = 20
pronostico_lluvia = False

# Regla: ¿Es una hora adecuada para regar?
def hora_adecuada():
    global hora
    return hora and hora < 10 or hora > 18

# Regla principal: ¿Cuándo se debe activar el sistema de riego?
def activar_riego():
    global humedad_suelo, pronostico_lluvia, hora
    return humedad_suelo == "baja" and pronostico_lluvia == False and hora_adecuada()

def estado_riego():
    if activar_riego():
        return "Activado"
    else:
        return "No activado"

# Regla para alerta de temperatura extrema
def alerta_calor():
    global temperatura
    return temperatura and temperatura >= 32

# Mensaje de recomendación
def recomendacion():
    if activar_riego() and alerta_calor():
        print("Alerta: Riego activado con temperatura alta. Considere riego nocturno o por goteo.")
    else:
        print("Sin recomendaciones especiales para el riego.")